src/components/ui/Animation.ts

```diff
@@ -175,9 +164,8 @@ export default function animateChart(stx, animationParameters) {
         const period = this.layout.periodicity;
         const interval = this.layout.interval;
         const timeUnit = this.layout.timeUnit;
-
-        function cb(quote, prevQuote, chartJustAdvanced) {
-            return function (newData) {
+        function cb(quote: any, prevQuote: any, chartJustAdvanced: any) {
```

Comment:
I can see that you put @ts-nocheck for this file but I still would like to suggest the following types for your consideration 😊:
animationParameters: { stayPut?: boolean, ticksFromEdgeOfScreen?: number, granularity?: TGranularity, tension: number | null },
Math (line 57): typeof Math,
chart (line 67 & 92): typeof CIQ.ChartEngine.Chart,
symbol (line 67): string,
params (line 67): { fromTrade?: boolean, finalClose?: boolean, animationEntry?: boolean, animationLastBar?: boolean },
params (line 73) - params?: { noCreateDataSet? boolean, noCleanupDates?: boolean, allowReplaceOHL?: boolean, bypassGovernor?: boolean, fillGaps?: boolean, secondarySeries?: string, deleteItems?: boolean, useAsLastSale?: boolean, aggregatedVolume?: boolean },
record (line 101): Record<string, number>,
quote (line 167): TQuote,
prevQuote: TQuote,
chartJustAdvanced: boolean